log-collector/pkg/scp/scp.go

```diff
@@ -0,0 +1,56 @@
+package scp
+
+import (
+	"fmt"
+
+	"github.com/pkg/sftp"
+	"golang.org/x/crypto/ssh"
+)
+
+type Config struct {
+	User            string
+	Host            string
+	Port            int32
+	IdentifyKeyFile string
+}
+
+type Scp struct {
+	conn   *ssh.Client
+	client *sftp.Client
+}
+
+func NewClient(config *Config) (*Scp, error) {
+	conn, err := newSSHClient(config)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := sftp.NewClient(conn)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Scp{
+		conn:   conn,
+		client: client,
+	}, nil
+}
+
+func (s *Scp) Close() {
+	s.client.Close()
+	s.conn.Close()
+}
+
+func (s *Scp) GetPodsFilePaths(pod string) ([]string, error) {
```

Comment:
It's not meant to be a used as an independent package. And hence the coupling with collector to access only pod/service logs